pkgs/development/misc/cppreference-doc/default.nix

```diff
@@ -0,0 +1,26 @@
+{ lib, stdenvNoCC, fetchurl }:
+stdenvNoCC.mkDerivation rec {
+  pname = "cppreference-doc";
+  version = "2022.07.30";
+
+  src = let
+    ver = lib.strings.stringAsChars (c: if c == "." then "" else c) version;
+  in fetchurl {
+    url = "https://github.com/PeterFeicht/${pname}/releases/download/v${ver}/html-book-${ver}.tar.xz";
+    sha256 = "sha256-cfFQA8FouNxaAMuvGbZICps+h6t+Riqjnttj11EcAos=";
+  };
+
+  buildCommand = ''
+    local outdir=$out/share/cppreference/doc/html
+    mkdir -p $outdir
```

Comment:
Since we are only unpacking this can entirely be done in fetchurl and postFetch.

Reply:
I was totally unaware of `postFetch`... I've done almost a full rewrite now. Please review again and let me know if it makes sense what I did.